src/main/java/net/minecraftforge/event/world/DimensionPreloadEvent.java

```diff
@@ -0,0 +1,35 @@
+package net.minecraftforge.event.world;
+
+import net.minecraft.world.DimensionType;
+import net.minecraftforge.common.DimensionManager;
+import net.minecraftforge.fml.common.eventhandler.Cancelable;
+import net.minecraftforge.fml.common.eventhandler.Event;
+import net.minecraftforge.common.MinecraftForge;
+
+/**
+ * Created by goreacraft on 17/03/03.
+ *
+ * This event is fired when DimensionManager is starting to load a new dimmension
+ * dimension 0 (Overworld) will not trigger this event
+ * {@link DimensionManager#initDimension(int)},
+ * <br>
+ * This event is {@link Cancelable}.<br>
```

Comment:
i am not part of forgeessentials dev team, i was looking to see how others found a solution for this problem, and i don't want to use another dependency when it can be done with a simple forge edit